sdks/wasm/package-update/download-packages.sh

```diff
@@ -16,30 +23,103 @@ do
+        ASPNETCORE="$2"
+        shift 2
+        ;;
+    -m|--mono_working_dir)
+        MONO="$2"
+        shift 2
+        ;;
+    --asp_remote_name)
+        ASP_REMOTE_NAME="$2"
+        shift 2
+        ;;
+     --mono_remote_name)
+        MONO_REMOTE_NAME="$2"
+        shift 2
+        ;;
     *)
         shift
         ;;
     esac
 done
 
-PACKAGE_PATH=$HOME/.nuget/packages/microsoft.aspnetcore.components.webassembly.runtime/$RUNTIME/tools/dotnetwasm
+echo $MONO
+
+if [ -z "$RUNTIME_VER" ]; then
+	echo "Error: runtime version required. Use -r"
+	exit 1
+fi
+
+if [ -z "$ASPNETCORE" ]; then
+    echo "Error: aspnetcore working directory required. Use -a"
+    exit 1
+fi
+
+if [ -z "$MONO" ]; then
+    echo "Error: mono working directory required. Use -m"
+    exit 1
+fi
+
+if  [ -z "$FILEPATH" -a -z "$URL" ]; then
+	echo "Error: Either path to the wasm package file required, or the url to download from."
+	exit 1
+fi
+
+NUGET_HOME=${NUGET_HOME:-"$HOME/.nuget"}
+PACKAGE_PATH="$NUGET_HOME/packages/microsoft.aspnetcore.components.webassembly.runtime/$RUNTIME_VER/tools/dotnetwasm"
+PROXY_PACKAGE_PATH="$NUGET_HOME/packages/microsoft.aspnetcore.components.webassembly.devserver/$RUNTIME_VER/tools/BlazorDebugProxy"
+ASP_PROXY_PATH="$ASPNETCORE/src/Components/WebAssembly/DebugProxy/src"
+MONO_PROXY_PATH="$MONO/sdks/wasm/Mono.WebAssembly.DebuggerProxy"
+TMP_DIR=`mktemp -d`
+TMP_PKG_DIR=$TMP_DIR/wasm-package/
+
+if [ ! -d "$NUGET_HOME" ]; then
+	echo "NUGET_HOME envar = $NUGET_HOME does not exist."
+	exit 1
+fi
 
 if [ -z "$FILEPATH" ]
 then
-    wget -O wasm-package.zip $URL
-    unzip wasm-package.zip -d wasm-package/
-    FILEPATH=wasm-package/
+    wget -O "$TMP_DIR/wasm-package.zip" $URL
+    unzip "$TMP_DIR/wasm-package.zip" -d $TMP_PKG_DIR
+else
+    unzip "$FILEPATH" -d $TMP_PKG_DIR
 fi
 
-rm -r $PACKAGE_PATH/bcl
-cp $FILEPATH/wasm-bcl $PACKAGE_PATH/bcl
+echo "pkg: $TMP_PKG_DIR"
 
-rm -r $PACKAGE_PATH/framework
-cp $FILEPATH/framework $PACKAGE_PATH/framework
+rm -r "$PACKAGE_PATH/bcl"
+cp -r "$TMP_PKG_DIR/wasm-bcl/wasm/" "$PACKAGE_PATH/bcl"
+
+rm -r "$PACKAGE_PATH/framework"
+cp -r "$TMP_PKG_DIR/framework" "$PACKAGE_PATH/framework"
 
 rm "$PACKAGE_PATH"/wasm/*
-cp $FILEPATH/builds/release/dotnet.$RUNTIME.js $PACKAGE_PATH/wasm/
-cp $FILEPATH/builds/release/dotnet.wasm $PACKAGE_PATH/wasm/
+cp "$TMP_PKG_DIR/builds/release/dotnet.js" "$PACKAGE_PATH/wasm/dotnet.${RUNTIME_VER}.js"
+cp "$TMP_PKG_DIR/builds/release/dotnet.wasm" "$PACKAGE_PATH/wasm/"
+
+echo "Replacing DebuggerProxy..."
+
+cd $MONO
+git checkout $MONO_BRANCH_NAME
+git reset --hard
+git fetch $MONO_REMOTE_NAME
+git pull $MONO_REMOTE_NAME $MONO_BRANCH_NAME 
+
+cd $ASPNETCORE 
+git checkout $ASP_BRANCH_NAME
+git reset --hard
+git fetch $ASP_REMOTE_NAME
+git pull $ASP_REMOTE_NAME $ASP_BRANCH_NAME
+
+./clean.sh || true
+
+cp $MONO_PROXY_PATH/*.cs "$ASP_PROXY_PATH/MonoDebugProxy/ws-proxy/"
+rm "$ASP_PROXY_PATH/MonoDebugProxy/ws-proxy/AssemblyInfo.cs"
+./build.sh || true
```

Comment:
Same for this.

Reply:
I was still getting build errors regarding to some python script, but it builds the Debug Proxy properly, which is all the we need anyway, so I'm choosing to ignore it. But yea if there is an error with building the Debug Proxy then that might be a problem

Reply:
What kinda use case is this script meant for? I'm wondering if this script is doing too much.

Reply:
I just need to build the DebuggerProxy, but that script builds all of aspnetcore.